### PR TITLE
Complete GUI Based Sudoku

### LINKED ISSUE #2: Building a GUI-Based Sudoku Solver in C with GTK
### To create a GUI-based Sudoku solver in C, use the GTK library. First, write your Sudoku-solving logic in C using a backtracking algorithm. Then, create the GUI by initializing a GTK application and designing a 9x9 grid of text entry widgets to represent the Sudoku board. Add a "Solve" button that triggers the solver function to update the grid with the solution. Use GTK's layout management to arrange the grid and button, and update the values in the entries after solving the puzzle. This structure provides an interactive Sudoku solver with a graphical interface.

### LINKED ISSUE #16: [ Enhancement ] File Structure and Code Refactoring / Formatting
Hello!
I was thinking to help in managing the file structure and refactoring of the code.

# Benefits
- This will greatly benefit in any future collaborations and QoL for the developers who will be potentially work on this project.
- Easier to manage the codebase.

I can show you some examples of how it will work out if you want.

Happy Hacking!!

### PR DESCRIPTION
**Pull request by nandit27**
Completely GUI introduced. Process of running is that install gtk3+ on your system and then run "make" command in terminal and then run ./sudoku_solver and GUI should appear.
@soni-shashan

### CODE REVIEW

**Comment by nandit27** [comment]
Screenshot 2024-10-24 at 5 32 14 PM
this is before solving the sudoku

Screenshot 2024-10-24 at 5 32 21 PM
this is after solving the sudoku

**Comment by soni-shashan** [comment]
@nandit27 well done,

**Comment by soni-shashan** [comment]
@nandit27 please add in readme.md section how it run in windows, Linux and macOS

**Comment by soni-shashan** [comment]
### I will merge it, and I am updating the README.md file.

File: src/gui/main.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#include "sudoku_gui.h"
+#include "sudoku.h"
+
+// Global masks from sudoku.h
+int rowMask[MAX_SIZE], colMask[MAX_SIZE], boxMask[MAX_SIZE];
```
Comment (soni-shashan) [inline src/gui/main.c]: ### this line shows error multiple declarations remove it

**Comment by nandit27** [comment]
@soni-shashan level 3 issue will have 8 pts and level 1 will have 3 pts so total 11 points should be added instead of only 8 pts.

**Comment by soni-shashan** [comment]
### @nandit27  I assign the label at PR Level 3, so 8 points will be added if I mention both Level 1 and Level 3. According to the Club Gamma backend algorithm, the points for the highest priority label are counted. If you have any issues regarding this, please contact the club lead, @jalaym825 , as you have addressed two issues in one PR.

**Comment by nandit27** [comment]
@jalaym825 is it possible to get pts of both the issues ? 😅

**Comment by jalaym825** [comment]
hey @nandit27 
gave you level 4 with 15 points. enjoy :)

**Pull request merged**